test-frame-common/src/main/java/io/skodjob/testframe/clients/KubeClient.java

```diff
@@ -98,6 +105,185 @@ public List<HasMetadata> readResourcesFromFile(InputStream is) throws IOExceptio
         }
     }
 
+    /**
+     * Check if namespace exists in current cluster
+     * @param namespace namespace name
+     * @return true if namespace exists
+     */
+    public boolean namespaceExists(String namespace) {
+        return client.namespaces().list().getItems().stream().map(n -> n.getMetadata().getName())
+                .toList().contains(namespace);
+    }
+
+    /**
+     * Create resources from file and apply modifier
+     * @param namespace dest namespace
+     * @param is stream of data
+     * @param modifier modifier method
+     * @throws IOException exception
+     */
+    public void create(String namespace, InputStream is, Function<HasMetadata, HasMetadata> modifier)
+            throws IOException {
+        try (is) {
+            client.load(is).get().forEach(i -> {
+                HasMetadata h = modifier.apply(i);
+                if (h != null) {
+                    if (client.resource(h).inNamespace(namespace).get() == null) {
+                        LOGGER.debug(LoggerUtils.RESOURCE_LOGGER_PATTERN,
+                                "Creating", h.getKind(), h.getMetadata().getName());
+                        client.resource(h).inNamespace(namespace).create();
+                    } else {
+                        LOGGER.debug(LoggerUtils.RESOURCE_LOGGER_PATTERN,
+                                "Updating", h.getKind(), h.getMetadata().getName());
+                        client.resource(h).inNamespace(namespace).update();
+                    }
+                }
+            });
+        }
+    }
+
+    /**
+     * Create resources from file and apply modifier
+     * @param is stream of data
+     * @param modifier modifier method
+     * @throws IOException exception
+     */
+    public void create(InputStream is, Function<HasMetadata, HasMetadata> modifier) throws IOException {
```

Comment:
Same here about the name (and that it should be split to `create` and `update` methods) and the redundancy.